addons/purchase/stock.py

```diff
@@ -31,6 +31,13 @@ class stock_move(osv.osv):
             readonly=True),
     }
 
+    def get_price_unit(self, cr, uid, move, context=None):
+        """ Returns the unit price to store on the quant """
+        if move.purchase_line_id:
+            return move.price_unit
```

Comment:
@nim-odoo Is not it more logical to use move.price_unit all the time then in the base method?  (I don't know why we kept having twice the same logic in fact)